phoenicis-repository/src/main/java/org/phoenicis/repository/types/GitRepository.java

```diff
@@ -65,72 +69,103 @@ private File createRepositoryLocation(String cacheDirectoryPath) {
         return new File(cacheDirectoryPath + "/git" + hashcode);
     }
 
-    @Override
-    public synchronized RepositoryDTO fetchInstallableApplications() {
-        RepositoryDTO result = null;
-        try {
-            mutex.acquire();
+    private void cloneOrUpdate() throws RepositoryException {
+        synchronized (mutex) {
             try {
                 LOGGER.info("Begin fetching process of " + this);
 
-                boolean folderExists = this.localFolder.exists();
+                boolean lockFileExists = this.lockFile.exists();
 
-                // check that the repository folder exists
-                if (!folderExists) {
-                    LOGGER.info("Creating local folder for " + this);
+                // check that the repository lock file exists
+                if (!lockFileExists) {
```

Comment:
When is the file deleted?

Reply:
Never. Why should it be deleted?

Reply:
So it is not a lock then, is it?

Reply:
No, the file itself is not the lock. I acquire a lock **for** the file with `java.nio.channels.FileLock` (on OS level).

Reply:
Ok, it seems that I have missed something